sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -270,11 +260,141 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
     keys. Enables resolution of Key Vault references in configuration settings.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, **kwargs) -> None:
         self._dict: Dict[str, str] = {}
         self._trim_prefixes: List[str] = []
         self._client: Optional[AzureAppConfigurationClient] = None
         self._secret_clients: Dict[str, SecretClient] = {}
+        self._key_vault_options: Optional[AzureAppConfigurationKeyVaultOptions] = kwargs.pop("key_vault_options", None)
+        self._selects: List[SettingSelector] = kwargs.pop(
+            "selects", [SettingSelector(key_filter="*", label_filter=EMPTY_LABEL)]
+        )
+        self._sentinel_keys: List[str] = []
+
+        trim_prefixes: List[str] = kwargs.pop("trim_prefixes", [])
+        self._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
+        self._configuration_refresh = self._ConfigurationRefresh(**kwargs)
+
+    def refresh_configuration_settings(self, func):
+        @wraps(func)
+        def refresh_wrapper(*args, **kwargs):
+            self.refresh()
+            return func(*args, **kwargs)
+
+        return refresh_wrapper
+
+    def refresh(self, **kwargs) -> None:
+        # pylint:disable=protected-access
+        refresh_registrations = self._configuration_refresh.refresh_options._refresh_registrations
+        if len(refresh_registrations) == 0:
+            logging.debug("Refresh called but no refresh options set.")
+            self._configuration_refresh.refresh_options._on_error()
+            return
+
+        if datetime.now() < self._configuration_refresh.next_refresh_time:
+            logging.debug("Refresh called but refresh interval not elapsed.")
+            return
+
+        try:
+            # pylint:disable=protected-access
+            for registration in refresh_registrations:
+                if registration.refresh_all:
+                    updated_sentinel = self._client.get_configuration_setting(
+                        key=registration.key_filter, label=registration.label_filter, **kwargs
+                    )
+                    if registration.etag != updated_sentinel.etag:
```

Comment:
Instead of checking the etag here, we can use conditional request to get key value only if it changed on the server.
https://github.com/Azure/azure-sdk-for-python/blob/edbff1d3ccdf1ba1da3a3b404656dc6fea613c75/sdk/appconfiguration/azure-appconfiguration/samples/conditional_operation_sample.py#L43